languages/java/reference/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,138 @@
-# How to implement an Java concept exercise
+# How to implement a TypeScript Concept Exercise
 
-TODO: describe how to implement a concept exercise for the Java track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes how to implement a Concept Exercise for the Java track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/reference/implementing-a-concept-exercise.md
+- `<slug>`: the name of the exercise in snake_case (e.g. `anonymous-methods`).
+- `<concepts>`: the Concepts the exercise is about (e.g. `loops`),
+- `<concept-1>`: a single Concept slug,
+- `<prerequisite-n>`: a single Concept slug,
+- `<uuid>`: a _new_ v4 UUID (random!)
+- `<first-and-last-name>`: your first and last name (e.g. `John Doe`)
+- `<git-email>`: the email address you use for git (e.g. `johndoe@email.com`)
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Having done this, please read the [Java Concept exercises introduction][concept-exercises]. If you have come up with something completely new, create a new issue _first_ so we can discuss the Concept Exercise.
+
+To implement a Concept Exercise, the following files must be created:
+
+<pre>
+languages
+└── java
+    └── exercises
+        └── concept
+            └── &lt;slug&gt;
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                ├── .gitignore
+                ├── .meta
+                    |── design.md
+                    |── config.json
+                    ├── example
+                        |── build.gradle
+                        └── src
+                            └── main
+                                └── java
+                                    └── &lt;slug&gt;.java
+                    ├── &lt;slug&gt;
+                        |── build.gradle
+                        └── src
+                            └── main
+                                └── java
+                                    └── &lt;slug&gt;.java
+                            └── test
+                                └── java
+                                    └── &lt;slug&gt;Test.java
```

Comment:
Is this section intended to be back up another level (i.e. directly in the concept/SLUG directory, rather than being nested in the .meta directory)? If so the indent needs correcting

Reply:
It should indeed be at the slug directory level

Reply:
good catch!